src/Okta.Idx.Sdk/OktaVerify/OktaVerifyAuthenticationOptions.cs

```diff
@@ -0,0 +1,253 @@
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Okta.Idx.Sdk.Helpers;
+
+namespace Okta.Idx.Sdk.OktaVerify
+{
+    /// <summary>
+    /// A class that contains the options for authentication with Okta Verify.
+    /// </summary>
+    public class OktaVerifyAuthenticationOptions
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OktaVerifyAuthenticationOptions"/> class.
+        /// </summary>
+        /// <param name="authenticationResponse">The authentication response.</param>
+        /// <param name="idxResponse">The Idx response.</param>
+        public OktaVerifyAuthenticationOptions(AuthenticationResponse authenticationResponse, IIdxResponse idxResponse)
+        {
+            this.AuthenticationResponse = authenticationResponse;
+            this.IdxContext = authenticationResponse.IdxContext;
+            this.StateHandle = idxResponse.StateHandle;
+            this.SelectAuthenticatorAuthenticateRemediationOption = idxResponse.FindRemediationOption(RemediationType.SelectAuthenticatorAuthenticate);
+            this.CurrentAuthenticator = idxResponse.GetProperty<IIdxResponse>("currentAuthenticator").GetProperty<AuthenticatorValue>("value");
+        }
+
+        /// <summary>
+        /// Gets or sets the authentication response.
+        /// </summary>
+        protected internal AuthenticationResponse AuthenticationResponse { get; set; }
+
+        /// <summary>
+        /// Gets the Idx context.
+        /// </summary>
+        protected IIdxContext IdxContext { get; }
+
+        /// <summary>
+        /// Gets or sets the state handle.
+        /// </summary>
+        public string StateHandle { get; set; }
+
+        /// <summary>
+        /// Gets the `select-authenticator-authenticate` remediation option.
+        /// </summary>
+        protected IRemediationOption SelectAuthenticatorAuthenticateRemediationOption { get; }
+
+        /// <summary>
+        /// Gets the `challenge-poll` remediation option.
+        /// </summary>
+        protected IRemediationOption ChallengePollRemediationOption { get; private set; }
+
+        /// <summary>
+        /// Gets the `challenge-authenticator` remediation option.
+        /// </summary>
+        protected IRemediationOption ChallengeAuthenticatorRemediationOption { get; private set; }
+
+        /// <summary>
+        /// Gets the current authenticator.
+        /// </summary>
+        protected internal IAuthenticatorValue CurrentAuthenticator { get; }
+
+        /// <summary>
+        /// Selects the specified authentication method type.
+        /// </summary>
+        /// <param name="methodType">The method type.</param>
+        /// <returns>The authentication response.</returns>
+        protected async Task<AuthenticationResponse> SelectOktaVerifyMethodAsync(string methodType)
```

Comment:
We should follow the existing design here where all the methods are called via the IdxClient.

Reply:
I'll move the implementation.